### PR TITLE
Build scripts

### PR DESCRIPTION
**Pull request by Fadiabb**

### CODE REVIEW

File: docs/install-server-certificate.md [hunk 1]
```diff
@@ -44,25 +44,21 @@ We recommend to
...
+<!-- MARKDOWN-AUTO-DOCS:START (CODE:src=../docs/scripts/TLSConfigsForITest.sh&lines=31-37) -->
+<!-- The below code snippet is automatically added from ../docs/scripts/TLSConfigsForITest.sh -->
+```sh
+        listen 443 ssl default_server; # ipv4
+        listen [::]:443 ssl http2 default_server;  # ipv6
 
-    ssl_certificate /etc/ssl/{domainName}.pem; # or bundle.crt
-    ssl_certificate_key /etc/ssl/{domainName}.key";
+       '${SSL_CERTIFICATE}' # e.g. ssl_certificate /etc/ssl/csaf/bundle.crt
+       '${SSL_CERTIFICATE_KEY}' # e.g. ssl_certificate_key /etc/ssl/csaf/testserver-key.pem;
```
Comment (bernhardreiter) [inline docs/install-server-certificate.md:57]: Same here
```
ssl_certificate_key '${SSL_CERTIFICATE_KEY}'
``` looks like being a bit more instructive

File: docs/client-certificate-setup.md [hunk 2]
```diff
@@ -5,29 +5,29 @@ a web browser.
...
-            # want to reveal if this location exists or not.
-            return 404;
-        }
-    }
-}
+adjust the content of the `server{}` block like shown in the following example:
+<!-- MARKDOWN-AUTO-DOCS:START (CODE:src=../docs/scripts/TLSClientConfigsForITest.sh&lines=25-40) -->
+<!-- The below code snippet is automatically added from ../docs/scripts/TLSClientConfigsForITest.sh -->
+```sh
+        ssl_client_certificate '${SSL_CLIENT_CERTIFICATE}' # e.g. ssl_client_certificate /etc/ssl/rootca-cert.pem;
```
Comment (tschmidtb51) [inline docs/client-certificate-setup.md:12]:
```suggestion
        ssl_client_certificate '${SSL_CLIENT_CERTIFICATE}'; # e.g. ssl_client_certificate /etc/ssl/rootca-cert.pem;

File: docs/client-certificate-setup.md [hunk 3]
```diff
@@ -5,29 +5,29 @@ a web browser.
...
+        ssl_verify_client optional;
+        ssl_verify_depth 2;
+
+        # This example allows access to all three TLP locations for all certs.
+        location ~ /.well-known/csaf/(red|green|amber)/{
```
Comment (tschmidtb51) [inline docs/client-certificate-setup.md:17]: As we have a new location directive don't we need to disable symlinks again?
```
        # For atomic directory switches
        disable_symlinks off;
```

File: docs/provider-setup.md [hunk 4]
```diff
@@ -89,26 +91,32 @@ Rename and place the `csaf_provider` binary file under `/usr/lib/cgi-bin/csaf_pr
...
-```
+<!-- MARKDOWN-AUTO-DOCS:START (CODE:src=../docs/scripts/setupProviderForITest.sh&lines=82-87) -->
+<!-- The below code snippet is automatically added from ../docs/scripts/setupProviderForITest.sh -->
+```sh
 # upload_signature = true
 # key = "/usr/lib/csaf/public.asc"
 key = "/usr/lib/csaf/private.asc"
 #tlps = ["green", "red"]
-canonical_url_prefix = "http://192.168.56.102"
+canonical_url_prefix = "https://localhost:8443"
```
Comment (tschmidtb51) [inline docs/provider-setup.md:101]: Why is there port 8443 in here?

File: docs/scripts/TLSClientConfigsForITest.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+set -e
+
+NGINX_CONFIG_PATH=/etc/nginx/sites-available/default
+
+cd ~/csaf_distribution/docs/scripts/
+source ./createCCForITest.sh
+
+echo '
+        ssl_client_certificate '${SSL_CLIENT_CERTIFICATE}' # e.g. ssl_client_certificate /etc/ssl/rootca-cert.pem;
```
Comment (tschmidtb51) [inline docs/scripts/TLSClientConfigsForITest.sh:25]: Same here missing `;`?

File: docs/scripts/TLSConfigsForITest.sh [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+## Create webserver cert
+source ./createWebserverCertForITest.sh
+
+# Configure nginx
+echo '
+        listen 443 ssl default_server; # ipv4
+        listen [::]:443 ssl http2 default_server;  # ipv6
+
+        ssl_certificate  '${SSL_CERTIFICATE}' # e.g. ssl_certificate /etc/ssl/csaf/bundle.crt
+        ssl_certificate_key '${SSL_CERTIFICATE_KEY}' # e.g. ssl_certificate_key /etc/ssl/csaf/testserver-key.pem;
```
Comment (tschmidtb51) [inline docs/scripts/TLSConfigsForITest.sh:35]: And here.

**Review by tschmidtb51 — COMMENTED** [review]
@bernhardreiter I had some remarks - could you please have a look whether they are valid?

File: docs/provider-setup.md (hunk 4 above)
Comment (tschmidtb51) [reply docs/provider-setup.md:101]: Sorry - found the place where it was set.

Comment (tschmidtb51) [reply docs/provider-setup.md:101]: Nevertheless, it looks weird when using `localhost:8443` here and all the other docs part refer to `192.168.56.102`...

File: docs/client-certificate-setup.md (hunk 2 above)
Comment (bernhardreiter) [reply docs/client-certificate-setup.md:12]: The variable has the semicolon, yes, should go outside. Thanks for spotting.
@Fadiabb please change this (for a new pull-request)

File: docs/client-certificate-setup.md (hunk 3 above)
Comment (bernhardreiter) [reply docs/client-certificate-setup.md:17]: Yes, I think so, though not being entirely sure, thanks for spotting.
@Fadiabb please add it.

**Comment by bernhardreiter** [comment]
@Fadiabb  for the documentation it should always be the same example url where the provider can be found. Can you check that this is the case and the localhost:8443 is not visible then.

**Pull request merged**